examples/eventsub/src/twitch.rs

```diff
@@ -0,0 +1,360 @@
+            online = online_exists,
+            "got existing subs"
+        );
+        drop(subs);
+        if !online_exists {
+            let request = CreateEventSubSubscriptionRequest::new();
+            let body = CreateEventSubSubscriptionBody::new(
+                StreamOnlineV1::broadcaster_user_id(config.broadcaster.id.clone()),
+                twitch_eventsub::Transport::webhook(
+                    website.clone(),
+                    sign_secret.secret_str().to_string(),
+                ),
+            );
+            if std::env::var("DEV").is_ok() {
+                tracing::info!("In dev mode, not registering eventsubs");
+            } else {
+                client
+                    .req_post(request, body, &*token.read().await)
+                    .await
+                    .wrap_err_with(|| "when registering online event")?;
+            }
+        }
+
+        if !offline_exists {
+            let request = CreateEventSubSubscriptionRequest::default();
+            let body = CreateEventSubSubscriptionBody::new(
+                StreamOfflineV1::broadcaster_user_id(config.broadcaster.id.clone()),
+                twitch_eventsub::Transport::webhook(
+                    website.clone(),
+                    sign_secret.secret_str().to_string(),
+                ),
+            );
+            if std::env::var("DEV").is_ok() {
+                tracing::info!("In dev mode, not registering eventsubs");
+                return Ok(());
+            } else {
+                client
+                    .req_post(request, body, &*token.read().await)
+                    .await
+                    .wrap_err_with(|| "when registering offline event")?;
+            }
+        }
+    }
+    #[allow(unreachable_code)]
+    Ok(())
+}
+
+pub async fn twitch_eventsub(
+    Extension(sender): Extension<Arc<watch::Sender<LiveStatus>>>,
+    Extension(opts): Extension<Arc<Opts>>,
+    Extension(config): Extension<Arc<Config>>,
+    Extension(cache): Extension<Arc<retainer::Cache<http::HeaderValue, ()>>>,
+    request: http::Request<axum::body::Body>,
+) -> impl IntoResponse {
+    const MAX_ALLOWED_RESPONSE_SIZE: u64 = 64 * 1024;
+
+    let (parts, body) = request.into_parts();
+    let response_content_length = match body.size_hint().upper() {
+        Some(v) => v,
+        None => MAX_ALLOWED_RESPONSE_SIZE + 1, /* Just to protect ourselves from a malicious response */
+    };
+    let body = if response_content_length < MAX_ALLOWED_RESPONSE_SIZE {
+        hyper::body::to_bytes(body).await.unwrap()
+    } else {
+        panic!("too big data given")
+    };
+
+    let request = http::Request::from_parts(parts, &*body);
+
+    tracing::debug!("got event {}", std::str::from_utf8(request.body()).unwrap());
+    tracing::debug!("got event headers {:?}", request.headers());
+    if !Event::verify_payload(&request, opts.sign_secret.secret()) {
+        return (StatusCode::BAD_REQUEST, "Invalid signature".to_string());
+    }
```

Comment:
In the long run, this should probably be an extractor/middleware.

A while ago I made extractors for both `actix-web` and `axum` [here](https://github.com/Nerixyz/twitch-eventsub-rs). Though the extractor for `axum` doesn't support multiple events, since I couldn't find the equivalent of [guards](https://docs.rs/actix-web/latest/actix_web/guard/index.html) for axum.